packages/eslint-plugin/__tests__/stylex-margin-padding.js

```diff
@@ -0,0 +1,262 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ *
+ */
+
+'use strict';
+
+jest.disableAutomock();
+
+const { RuleTester: ESLintTester } = require('eslint');
+const rule = require('../src/stylex-margin-padding-shorthand');
+
+const eslintTester = new ESLintTester({
+  parser: require.resolve('hermes-eslint'),
+  parserOptions: {
+    ecmaVersion: 6,
+    sourceType: 'module',
+  },
+});
+
+eslintTester.run('stylex-sort-keys', rule.default, {
+  valid: [
+    {
+      code: `
+      import stylex from 'stylex';
+      const styles = stylex.create({
+        main: {
+          marginInlineEnd: '14px',
+          marginInlineStart: '14px',
+        },
+      })
+    `,
+    },
+  ],
+  invalid: [
+    {
+      code: `
+        import stylex from 'stylex';
+        const styles = stylex.create({
+          main: {
+            margin: '10px 12px 13px 14px',
```

Comment:
In practice we can't really tell what the author intent was here, since 'margin' is a shorthand for physical directions. The safe expansion would be to marginTop, marginLeft, etc. But typically we want to use logical directions.

Reply:
@nmn what is stylex currently doing under the hood when it receives values like this? are the left/right values being converted to logical styles?

Reply:
This is the correct expansion based on what StyleX does internally:

```
marginTop: '10px',
marginInlineEnd: '12px',
marginBottom: '13px',
marginInlineStart: '14px',
```

Legacy Haste reasons....

Reply:
I think for the purposes of the eslint plugin, we have to expand this correctly to physical properties instead of propagating what is essentially an internal StyleX bug to OSS. It means that internally we'll either have to patch the OSS eslint plugin to do the wrong thing, or better yet have a codemod do this transform and replicate the legacy behavior of StyleX. Then have the semantically correct eslint suggestion/auto-fix will work internally going forward too.